packages/utils/src/facility/bulk-import-facility.ts

```diff
@@ -141,15 +126,67 @@ async function main({ cxId, inputPath, dryrun }: FacilityImportParams) {
     readFileFromLocal(inputPath, parser).catch(reject);
   });
 
-  await Promise.all(rowPromises);
+  return rows;
+}
 
-  await fs.writeFile(payloadCreatesFilePath, JSON.stringify(createdFacilities, null, 2), "utf8");
+async function processRow(
+  row: InputRowFacilityImport,
+  cxId: string,
+  isDryRun: boolean,
+  logsFilePath: string
+): Promise<FacilityInternalDetails | null> {
+  let rowSuccess = true;
+  let message: string | undefined = undefined;
 
-  console.log(
-    `############## FINISHED AT: ${new Date().toISOString()}  ${
-      isDryRun ? "[DRY RUN]" : ""
-    } ##############`
-  );
+  try {
+    const npiFacility = await getFacilityByNpiOrFail(row.npi);
+
+    const params = {
+      ...row,
+      cqActive,
+      cwActive,
+    };
+
+    const metriportFacility = translateNpiFacilityToMetriportFacility(npiFacility, params);
+    const otherNames = npiFacility.other_names ?? [];
+    if (otherNames.length > 0) {
+      if (
+        !facilityNamesMatch(
+          npiFacility.other_names[0].organization_name,
+          metriportFacility.nameInMetriport
+        )
+      ) {
+        throw new Error(
+          `Name mismatch: Registry='${npiFacility.other_names[0].organization_name}', CSV='${metriportFacility.nameInMetriport}'`
+        );
+      }
+    }
+
+    const existingFacility = await getFacilityByNpi(cxId, row.npi);
+    if (existingFacility) {
+      throw new Error(
+        `Can't create a new facility with the same NPI as facility with ID: ${existingFacility.id} and name: ${existingFacility.name}`
+      );
+    }
+    if (!isDryRun) {
+      await createFacility(metriportFacility, cxId);
+    }
+    console.log(`Successfully created facility with npi: ${row.npi}`);
+    return metriportFacility;
+  } catch (err: unknown) {
+    rowSuccess = false;
+    if (axios.isAxiosError(err) && err.response?.status === 400) {
+      message = err.response.data?.detail ?? err.response.data?.title ?? err.message;
+      console.log(message);
+    } else {
+      console.log(err);
+      message = errorToString(err);
+    }
+    return null;
+  } finally {
+    await writeToCsv(logsFilePath, rowSuccess, message, row);
+    await sleep(timeout.asMilliseconds());
```

Comment:
Let's keep the wait for each iteration on the controller of the iteration/loop, not on the iteration/function itself, please.